File: actin-common/src/main/java/com/hartwig/actin/clinical/datamodel/TumorDetails.java
package com.hartwig.actin.clinical.datamodel;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class TumorDetails {

    @Nullable
    public abstract String primaryTumorLocation();

    @Nullable
    public abstract String primaryTumorSubLocation();

    @Nullable
    public abstract String primaryTumorType();

    @Nullable
    public abstract String primaryTumorSubType();

    @Nullable
    public abstract String primaryTumorExtraDetails();

    @Nullable
    public abstract Set<String> doids();

    @Nullable
    public abstract TumorStage stage();

    @Nullable
    public abstract Boolean hasMeasurableLesionRecist();

    @Nullable
    public abstract Boolean hasBrainLesions();

    @Nullable
    public abstract Boolean hasActiveBrainLesions();

    @Nullable
    public abstract Boolean hasSymptomaticBrainLesions();

    @Nullable
    public abstract Boolean hasCnsLesions();

    @Nullable
    public abstract Boolean hasActiveCnsLesions();

    @Nullable
    public abstract Boolean hasSymptomaticCnsLesions();

    @Nullable
    public abstract Boolean hasBoneLesions();

    @Nullable
    public abstract Boolean hasLiverLesions();

    @Nullable
    public abstract Boolean hasOtherLesions();

    @Nullable
    public abstract String otherLesions();

}

File: common/src/main/kotlin/com/hartwig/actin/clinical/datamodel/TumorDetails.kt
package com.hartwig.actin.clinical.datamodel

data class TumorDetails(
    val primaryTumorLocation: String? = null,
    val primaryTumorSubLocation: String? = null,
    val primaryTumorType: String? = null,
    val primaryTumorSubType: String? = null,
    val primaryTumorExtraDetails: String? = null,
    val doids: Set<String>? = null,
    val stage: TumorStage? = null,
    val hasMeasurableDisease: Boolean? = null,
    val hasBrainLesions: Boolean? = null,
    val brainLesionsCount: Int? = null,
    val hasActiveBrainLesions: Boolean? = null,
    val hasCnsLesions: Boolean? = null,
    val cnsLesionsCount: Int? = null,
    val hasActiveCnsLesions: Boolean? = null,
    val hasBoneLesions: Boolean? = null,
    val boneLesionsCount: Int? = null,
    val hasLiverLesions: Boolean? = null,
    val liverLesionsCount: Int? = null,
    val hasLungLesions: Boolean? = null,
    val lungLesionsCount: Int? = null,
    val hasLymphNodeLesions: Boolean? = null,
    val lymphNodeLesionsCount: Int? = null,
    val otherLesions: List<String>? = null,
    val biopsyLocation: String? = null
)
